### PR TITLE
Udpate ts typings

### LINKED ISSUE #1114: [react-jss] withStyles typings should take strings as object values
__Is your feature request related to a problem?__
`withStyles` gives a type error when taking a string as object value despite `jss-plugin-template` being a default plugin:

```javascript
withStyles({
  root: `font-size: 1em;`
});
```

__Describe the solution you'd like__
`withStyles` should not error out when taking strings as object values. This should also be taken into account for the new hooks api.

__Are you willing to implement it?__
Yes

**Edit:** It looks like the culprit is `jss` exported `Style`: `export type Style = { [key: string]: Style | any }`.

### LINKED ISSUE #1132: createUseStyles typescript declaration missing
__Steps to reproduce:__

1. Create a fresh react app
  - `npx create-react-app use-styles-ts`
  - `cd use-styles-ts`
2. Add react-jss and typescript
  - `yarn add react-jss@next typescript`
  - `yarn add -D @types/react`
3. Rename `App.js -> App.tsx` and implement a simple usage of `createUseStyles` hook
```tsx
import React from 'react';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  App: { textAlign: 'center' },
})

function App() {
  const classes = useStyles();
  return (
    <div className={classes.App}>
    ...
```
4. Start app
  - `yarn start`

__Expected behavior:__
- App compiles and runs normally

__Describe the bug:__
- App fails to compile with error:
```bash
Failed to compile.

/Users/robcmills/src/jss-use-styles/src/App.tsx
TypeScript error in /Users/robcmills/src/jss-use-styles/src/App.tsx(2,10):
Module '"../../../../../Users/robcmills/src/jss-use-styles/node_modules/react-jss/src"' has no exported member 'createUseStyles'.  TS2305

    1 | import React from 'react';
  > 2 | import { createUseStyles } from 'react-jss';
      |          ^
    3 | import logo from './logo.svg';
    4 | import './App.css';
```
- Apparently a typescript declaration for `createUseStyles` is missing from `react-jss/src/index.d.ts`
- I have confirmed that adding a declaration for `createUseStyles` to this file silences the error

__Versions:__
 - react-jss: 10.0.0-alpha.17
 - typescript: 3.5.2
 - Browser [e.g. chrome, safari]: Chrome Version 74.0.3729.131 (Official Build) (64-bit)
 - OS [e.g. Windows, macOS]: macOS Sierra version 10.12.6

### PR DESCRIPTION
**Pull request by HenriBeck**
__What would you like to add/fix?__
Update TS types

__Corresponding issue (if exists):__ 
Closes #1132
Closes #1114

### CODE REVIEW

File: packages/react-jss/src/index.d.ts [hunk 1]
```diff
@@ -51,9 +53,28 @@ interface Options extends StyleSheetFactoryOptions {
 
 type Omit<T, K> = Pick<T, Exclude<keyof T, K>>
 
+export {Styles}
+
+declare function createUseStyles<C extends string>(
+  styles: Record<C, Style>,
```
Comment (HenriBeck) [inline packages/react-jss/src/index.d.ts]: The inference here doesn't work yet (Not sure why though, at least in Webstorm I couldn't get it working)

File: packages/jss/package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.3.1",
+    "csstype": "^2.6.5",
```
Comment (kof) [inline packages/jss/package.json]: did you really mean it as a direct dependency?

Comment (HenriBeck) [reply packages/jss/package.json]: Yes, most of the time, type dependency should be direct ones. Only in individual cases they should  a peer one (React for example)

**Comment by kof** [comment]
don't forget the changelog

**Comment by peku33** [comment]
Any update on this?

**Pull request merged**